File: stripe/src/main/java/com/stripe/android/exception/APIException.java
package com.stripe.android.exception;

import androidx.annotation.Nullable;

import com.stripe.android.StripeError;

/**
 * An {@link Exception} that represents an internal problem with Stripe's servers.
 */
public class APIException extends StripeException {

    public APIException(@Nullable String message, @Nullable String requestId,
                        @Nullable Integer statusCode, @Nullable StripeError stripeError,
                        @Nullable Throwable e) {
        super(stripeError, message, requestId, statusCode, e);
    }
}

File: stripe/src/main/java/com/stripe/android/exception/APIException.kt
package com.stripe.android.exception

import com.stripe.android.StripeError

/**
 * An [Exception] that represents an internal problem with Stripe's servers.
 */
class APIException(
    stripeError: StripeError? = null,
    requestId: String? = null,
    statusCode: Int = 0,
    message: String? = stripeError?.message,
    cause: Throwable? = null
) : StripeException(
    stripeError = stripeError,
    requestId = requestId,
    statusCode = statusCode,
    cause = cause,
    message = message
) {
    internal constructor(throwable: Throwable) : this(
        message = throwable.message,
        cause = throwable
    )
}
